File: pilhas-filas/python/queue_test.py
from collections import deque

queue = deque()

print(f"queue.size(): {len(queue)}")
print(f"queue.isEmpty(): {len(queue) == 0}")

queue.append("Maria")
queue.append("Joao")
queue.append("Ana")
print("Adicionados: Maria, Joao, Ana")

print(f"queue.size(): {len(queue)}")
print(f"queue.isEmpty(): {len(queue) == 0}")
print(f"queue.peek(): {queue[0]}")

print("Itens:")
while queue:
    print(queue.popleft())

File: pilhas-filas/javascript/queue-test.js
let queue = [];

console.log(`queue.size(): ${queue.length}`);
console.log(`queue.isEmpty(): ${queue.length === 0}`);

queue.push("Maria");
queue.push("Joao");
queue.push("Ana");
console.log("Adicionados: Maria, Joao, Ana");

console.log(`queue.size(): ${queue.length}`);
console.log(`queue.isEmpty(): ${queue.length === 0}`);
console.log(`queue.peek(): ${queue[0]}`);

console.log("Itens:");
while (queue.length > 0) {
    console.log(queue.shift()); // Usando shift para remover o primeiro elemento
}
